dev/com.ibm.ws.st.liberty.buildplugin.integration/src/com/ibm/ws/st/liberty/buildplugin/integration/internal/AbstractCustomServerVariablesHandler.java

```diff
@@ -71,23 +68,23 @@ public void addCustomServerVariables(ConfigVars configVars, IProject project) {
             // Load variables from server.env file (when applicable)
             String serverEnvFile = libertyBuildProjectConfiguration.getConfigValue(ConfigurationType.serverEnv);
             if (serverEnvFile != null) {
-                URI serverEnvFileURI;
-                serverEnvFileURI = PathUtil.getURIForFilePath(serverEnvFile);
-                if (serverEnvFileURI != null) {
-                    DocumentLocation serverEnvFileLocation = DocumentLocation.createDocumentLocation(serverEnvFileURI, DocumentLocation.Type.SERVER_ENV);
-                    loadPropertiesFile(configVars, serverEnvFile, serverEnvFileLocation);
+                try {
+                    ServerEnv serverEnv = new ServerEnv(new File(serverEnvFile), null);
```

Comment:
You're always reading the file even if the file hasn't changed?

Reply:
It looks like this is how the custom server variables manager was designed.  If we find it causes a major performance problem it can be addressed later.  These are usually very small files.

Reply:
ok